<launch>

    <arg name="only_agent_specific"    default="true"/>

    <!-- Enable custom logger configuration-->
    <env name="ROS_PYTHON_LOG_CONFIG_FILE" value="$(find tub_contest)/launch/python_logging.conf"/>

    <!--Infrastructure Nodes-->
    <include file="$(find tub_contest)/launch/graphhopper.launch">
        <arg name="initial_port" value="8770" />
    </include>

    <node pkg="knowledge_base" type="knowledge_base_node.py" name ="knowledgeBase_tub_contest" />


    <!--Agent nodes-->

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch">
        <arg name="agent_name" value="agentA1" />
        <!--The first bridge will always publish all general topics anyhow-->
        <arg name="only_agent_specific" value="false" />
    </include>
    <include file="$(find tub_contest)/launch/rhbp_agent.launch">
        <arg name="agent_name" value="agentA1" />
    </include>

    <!--Uncomment following blocks to use more than one agent-->

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch">
        <arg name="agent_name" value="agentA2" />
        <arg name="only_agent_specific" value="$(arg only_agent_specific)" />
    </include>
    <include file="$(find tub_contest)/launch/rhbp_agent.launch">
        <arg name="agent_name" value="agentA2" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch">
        <arg name="agent_name" value="agentA3" />
        <arg name="only_agent_specific" value="$(arg only_agent_specific)" />
    </include>
    <include file="$(find tub_contest)/launch/rhbp_agent.launch">
        <arg name="agent_name" value="agentA3" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch">
        <arg name="agent_name" value="agentA4" />
        <arg name="only_agent_specific" value="$(arg only_agent_specific)" />
    </include>
    <include file="$(find tub_contest)/launch/rhbp_agent.launch">
        <arg name="agent_name" value="agentA4" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch">
        <arg name="agent_name" value="agentA5" />
        <arg name="only_agent_specific" value="$(arg only_agent_specific)" />
    </include>
    <include file="$(find tub_contest)/launch/rhbp_agent.launch">
        <arg name="agent_name" value="agentA5" />
    </include>

    <include file="$(find mac_ros_bridge)/launch/mac_ros_bridge.launch">
        <arg name="agent_name" value="agentA6" />
        <arg name="only_agent_specific" value="$(arg only_agent_specific)" />
    </include>
    <include file="$(find tub_contest)/launch/rhbp_agent.launch">
        <arg name="agent_name" value="agentA6" />
    </include>




</launch>
